Extension/src/LanguageServer/copilotProviders.ts

```diff
@@ -38,19 +39,51 @@ export async function registerRelatedFilesProvider(): Promise<void> {
 
                         const getIncludesHandler = async () => (await getIncludesWithCancellation(1, token))?.includedFiles.map(file => vscode.Uri.file(file)) ?? [];
                         const getTraitsHandler = async () => {
-                            const chatContext: ChatContextResult | undefined = await (getActiveClient().getChatContext(token) ?? undefined);
+                            const cppContext = await getProjectContext(context, token);
 
-                            if (!chatContext) {
+                            if (!cppContext) {
                                 return undefined;
                             }
 
                             let traits: CopilotTrait[] = [
-                                { name: "language", value: chatContext.language, includeInPrompt: true, promptTextOverride: `The language is ${chatContext.language}.` },
-                                { name: "compiler", value: chatContext.compiler, includeInPrompt: true, promptTextOverride: `This project compiles using ${chatContext.compiler}.` },
-                                { name: "standardVersion", value: chatContext.standardVersion, includeInPrompt: true, promptTextOverride: `This project uses the ${chatContext.standardVersion} language standard.` },
-                                { name: "targetPlatform", value: chatContext.targetPlatform, includeInPrompt: true, promptTextOverride: `This build targets ${chatContext.targetPlatform}.` },
-                                { name: "targetArchitecture", value: chatContext.targetArchitecture, includeInPrompt: true, promptTextOverride: `This build targets ${chatContext.targetArchitecture}.` }
+                                { name: "intellisense", value: 'intellisense', includeInPrompt: true, promptTextOverride: `IntelliSense is currently configured with the following compiler information. It reflects the active configuration, and the project may have more configurations targeting different platforms.` },
```

Comment:
Should this name be more descriptive? Like "intelliSenseDisclaimer"?

Reply:
You had concerns that providing Copilot build information derived from IntelliSense could mislead it, and you suggested that we should make it clear that the information is from IntelliSense explicitly. These `intellisense`, `intellisenseBegin`, `intellisenseEnd` are traits that we can enable to address your concerns, see the following screenshot.

![image](https://github.com/user-attachments/assets/45b81ff8-4a01-4208-8e46-97a9ee960a37)

Reply:
Will change the name, thanks. `Value` is not used when `promptTextOverride` is defined. A trait with only the name and value looks like "name: value" in the prompts.